### PR TITLE
adding refetch parameter to ProductContext

### PR DESCRIPTION
**Pull request by ccamargo2294**
#### What is the purpose of this pull request?

We added the refetch parameter in ProductContext.js for then use in the ProductDetail component. 

#### What problem is this solving?

With this we can update the product information when some change in the catalog happens.

#### How should this be manually tested?

[Workspace](https://vtexexito--prueba1.myvtex.com/aguardiente/p)
NOTE: it is a exito subaccount and the workspace is free the custom components.

The `ProductQuery` prop in the `ProductDetails` component should have the method `refetch`. When this method is executed should update the product information.

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by luciannojunior** [comment]
Add a CHANGELOG.md describing your changes

File: react/ProductContext.js [hunk 1]
```diff
@@ -89,6 +89,7 @@ class ProductContext extends Component {
       params,
       params: { slug },
       client,
+      catalog : { refetch },
```
Comment (luciannojunior) [inline react/ProductContext.js]: I've tested this `refetch` directly and it didn't work. Did you manage to get it working?

Comment (ccamargo2294) [reply react/ProductContext.js]: I tested it and works but we should to implement a loading mask when this method is executed in the ProductDetail component.

Comment (ccamargo2294) [reply react/ProductContext.js]: The refetch method comes in the ProductQuery prop, i tested with the same flow in the SearchResult and it update the product information.

image

Comment (luciannojunior) [reply react/ProductContext.js]: If I'm not mistake, `productQuery` doesn't "exist", but `catalogQuery`. Did you verify if this `loading` comes from it (`catalogQuery`)?

**Comment by luciannojunior** [comment]
Merged with [#227](https://github.com/vtex-apps/store/pull/227)

**Pull request closed without merging**